test/functional/workgroup/tests/test-workgroup-Unordered-Single.hpp

```diff
@@ -92,10 +95,11 @@ void testWorkGroupUnorderedSingle(IndexType begin, IndexType end)
 
   WorkSite_type site = group.run();
 
-  working_res.memcpy(check_array, working_array, sizeof(IndexType) * N);
+  auto e = resource_type::get_default().get_event();
+  e.wait();
```

Comment:
Are we leaving it up to the user to to pull out the resource type like this when one isn't provided in the run, this seems like it could be confusing.

Reply:
Good point, this usage pattern is not something that a user should have to do, I did this more to check that the default resource was used when no resource was passed to run. We could have the site have a copy of the resource instead.

Reply:
That would be better. I noticed the get_event method was removed, do we still want to have a way of returning an event from run in some way or is that being delegated to users to handle manually.

Reply:
If we don't make an event during run then it may be more clear what's going on if you can only get the resource than if you could also get an event. So I'll add the resource for now.

Reply:
Sounds good to me